### DISCUSSION TITLE
[Bug] One File Per Command

### DISCUSSION THREAD

**Discussion by aytekinar** [fastapi/typer#1094] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the Typer documentation, with the integrated search.
- [X] I already searched in Google "How to X in Typer" and didn't find any information.
- [X] I already read and followed all the tutorials in the docs and didn't find an answer.
- [X] I already checked if it is not related to Typer but to [Click](https://github.com/pallets/click).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
import typer

child_app = typer.Typer()

@child_app.command()
def child():
    print("Child")

parent_app = typer.Typer()
# parent_app.add_typer(child_app, name="child", help="This is the child app")
parent_app.add_typer(child_app, help="This is the child app")

if __name__ == "__main__":
    parent_app()
```

### Description

Before `v0.15.0`, we did not have the so-called [One File Per Command](https://typer.tiangolo.com/tutorial/one-file-per-command/) capability, i.e., grouping of sub commands/typers without a name in a parent typer.

However, this came with a surprise. Now, I cannot get my `help` printed on `parent` typers.

Please see the above example code and uncomment the corresponding line and comment-out the following line to see the difference for yourself.

### Operating System

Linux

### Typer Version

0.15.1

### Python Version

Python 3.11.2

**Comment by YuriiMotov** [comment]
I don't think this is a bug..
Imagine your child app has more than 1 commands. For which command this help message should be used?

You can specify help message as a docstring of command or as a `help` parameter of `command` decorator:
```py
import typer

child_app = typer.Typer()

@child_app.command()
def child():
    """
    child command help message
    """
    print("Child")

parent_app = typer.Typer()

parent_app.add_typer(child_app)

if __name__ == "__main__":
    parent_app()
```
